fsharp-backend/src/LibExecution/StdLib/LibList.fs

```diff
@@ -318,27 +318,23 @@ let fns : List<BuiltInFn> =
       sqlSpec = NotYetImplementedTODO
       previewable = Pure
       deprecated = NotDeprecated }
-    //   ; { name = fn "List" "flatten" 0
-//     ; parameters = [Param.make "list" TList ""]
-//     ; returnType = TList
-//     ; description =
-//         "Returns a single list containing the values of every list directly in `list` (does not recursively flatten nested lists)."
-//     ; fn =
-//           (function
-//           | _, [DList l] ->
-//               let f a b =
-//                 match (a, b) with
-//                 | DList a, DList b ->
-//                     DList (List.append a b)
-//                 | _ ->
-//                     RT.error (DList [a; b]) "Flattening non-lists"
-//               in
-//               List.fold (DList []) ~f l
-//           | _ ->
-//               incorrectArgs ())
-//     ; sqlSpec = NotYetImplementedTODO
-//     ; previewable = Pure
-//     ; deprecated = NotDeprecated }
+    { name = fn "List" "flatten" 0
+      parameters = [Param.make "list" (TList (TList varA)) ""]
+      returnType = TList varA
+      description = "Returns a single list containing the values of every list directly in `list` (does not recursively flatten nested lists)."
+      fn =
+        (function
+        | _, [ DList l ] ->
+          let f a b =
+            match (a, b) with
+            | DList a, DList b -> DList (List.append a b)
+            | _ -> Errors.throw "Flattening non-lists"
+          in
+          Value(List.fold f (DList []) l)
```

Comment:
```suggestion
          let f list dv =
            match dv with
            | DList l -> List.append list l
            | _ -> Errors.throw "Flattening non-lists"
          in
          Value(DList(List.fold f (DList []) l))
```

Reply:
Was there a reason to put a DList in the accumulator/state, vs just a list. Comment above has a fix to avoid creating a DList each time; not tested though.

Reply:
No, just an oversight. Fixed in the latest version